lib/elixir/lib/enum.ex

```diff
@@ -3821,18 +3821,23 @@ defmodule Enum do
 end
 
 defimpl Enumerable, for: List do
+  def count([]), do: {:ok, 0}
   def count(_list), do: {:error, __MODULE__}
+
+  def member?([], _value), do: {:ok, false}
   def member?(_list, _value), do: {:error, __MODULE__}
+
+  def slice([]), do: {:ok, 0, fn _, _ -> [] end}
```

Comment:
@josevalim  Is there any reason to not to implement the 3 optimization introduced above?

Reply:
Can you show a code path where they would be actually invoked?

Reply:
The use case is the implementation of the Enumerable protocol. 
for instance,
- https://github.com/jimsynz/vivid.ex/blob/558e42f8e20ca892800209e712545c832170ef3a/lib/enumerable/vivid/buffer.ex#L33
- https://github.com/kennyp/checkex/blob/b94adf93e20abd5d9a30bb6bb2cb5b7f08d43991/lib/checkex.ex#L80-L81